### PR TITLE
DISM - turn on windows features, but skip all dependencies

### PR DESCRIPTION
**Pull request by chrisbednarski**
When trying to turn on the FTP service only, I get multiple other IIS features turning on

similar to /All flag being used with dism.exe
`dism.exe /Online /NoRestart /English /Enable-Feature /FeatureName:IIS-FTPExtensibility /All`

Added `EnableAll` property to the dism CA, so the value of the /All flag can be passed to [`DismEnableFeature`](https://learn.microsoft.com/en-us/windows-hardware/manufacture/desktop/dism/dismenablefeature-function)

Added `Order` property so the required prerequisite features can be enabled in the correct order

```
<PanelSW:Dism EnableFeature="IIS-WebServerRole" EnableAll="no" Order="1"/>
<PanelSW:Dism EnableFeature="IIS-FTPServer" EnableAll="no" Order="3"/>
<PanelSW:Dism EnableFeature="IIS-FTPSvc" EnableAll="no" Order="2"/>
<PanelSW:Dism EnableFeature="IIS-FTPExtensibility" EnableAll="no" Order="4"/>
```

### CODE REVIEW

File: Dism/Dism.cpp [hunk 1]
```diff
@@ -83,21 +85,27 @@ extern "C" UINT __stdcall Dism(MSIHANDLE hInstall)
...
 
 		hr = WcaReadIntegerFromCaData(&szCAD, (int*)&currState.eErrorHandling);
-		ExitOnFailure(hr, "Failed getting CustomActionData field");
+		ExitOnFailure(hr, "Failed getting ErrorHandling CustomActionData field");
+
+		hr = WcaReadIntegerFromCaData(&szCAD, (int*)&currState.bEnableAll);
+		ExitOnFailure(hr, "Failed getting EnableAll CustomActionData field");
 
-		hr = MemInsertIntoArray((void**)&pStates, 0, 1, dwStateNum, sizeof(ProgressReportState), 1);
+		hr = WcaReadIntegerFromCaData(&szCAD, (int*)&currState.nOrder);
```
Comment (nirbar) [inline Dism/Dism.cpp]: Is there any use for the Order field here?
Usually I only use it in the ORDER BY clause of the SQL statement

File: PanelSwCustomActions/DismSched.cpp [hunk 2]
```diff
@@ -78,6 +78,10 @@ extern "C" UINT __stdcall DismSched(MSIHANDLE hInstall)
 		ExitOnFailure(hr, "Failed to get Cost.");
 		hr = WcaGetRecordInteger(hRecord, 7, &nErrorHandling);
 		ExitOnFailure(hr, "Failed to get ErrorHandling.");
+		hr = WcaGetRecordInteger(hRecord, 8, &nEnableAll);
+		ExitOnFailure(hr, "Failed to get EnableAll.");
+		hr = WcaGetRecordInteger(hRecord, 9, &nOrder);
```
Comment (nirbar) [inline PanelSwCustomActions/DismSched.cpp]: Same comment as previous one- is there any use for the Order field other than in the ORDER BY clause?

**Review by nirbar — CHANGES_REQUESTED** [review]
@chrisbednarski Thanks for the contribution

File: PanelSwCustomActions/DismSched.cpp (hunk 2 above)
Comment (PestCoderNever) [reply PanelSwCustomActions/DismSched.cpp:83]: I will remove it

**Comment by chrisbednarski** [comment]
The order column was not being used
I've removed it

File: PanelSwCustomActions/DismSched.cpp (hunk 2 above)
Comment (chrisbednarski) [reply PanelSwCustomActions/DismSched.cpp]: removed

**Comment by chrisbednarski** [comment]
In your experience, will reversing the order of how the features are enabled affect the progress reporting / callback?
I haven't looked at progress reporting in depth after changing the order.

**Comment by nirbar** [comment]
@chrisbednarski I have no idea if the change will affect the progress reporting

**Comment by nirbar** [comment]
@chrisbednarski Can this PR be abandoned?

**Comment by chrisbednarski** [comment]
It sure can, thanks.

**Pull request closed without merging**